server/controllers/tripController.js

```diff
@@ -60,6 +60,38 @@ class TripController {
     util.setSuccess(200, message, data)
     return util.send(res);
   }
+
+  static async updateOpenTripDetails (req, res){
+    
+    const newTripData = {
+        origin: req.body.origin || req.oldTripDetails.origin,
```

Comment:
Yes, We need to use old trip details because on updating the trip, I first check if the cities provided are supported, I used `old trip details` so that In case the user did not *input* the `cities`, then the functionality would continue to run with no problem.

Reply:
@irfiacre I think there's no city verification happening in the **updateOpenTripDetails** . Another thing is that if no cities provided in the request, that means the user does not want to update the cities.